### PR TITLE
Adjust pipeline to new interface

### LINKED ISSUE #11: Update Datacomp example with new dataset and pipeline interface

### PR DESCRIPTION
**Pull request by mrchtr**
Fixes #11

### CODE REVIEW

File: src/components/add_clip_score/src/main.py [hunk 1]
```diff
@@ -28,13 +28,12 @@ class AddClipScore(PandasTransformComponent):
...
-    def __init__(self, *args) -> None:
+    def __init__(self, *args, **kwargs) -> None:
         pass
 
     def transform(self, dataframe: pd.DataFrame) -> pd.DataFrame:
         logger.info("Adding CLIP score...")
-
         dataframe["image_text_clipl14score"] = dataframe.apply(
-            lambda x: compute_clip_score(x["embeddings"], x["text_embedding"]))
+            lambda x: compute_clip_score(x["embedding"], x["text_embedding"]), axis=1)
```
Comment (RobbeSneyders) [inline src/components/add_clip_score/src/main.py]: How come this axis had to change?

File: src/components/detect_text/src/main.py [hunk 2]
```diff
@@ -160,11 +161,11 @@ def init_model(self, checkpoint_path):
         return self.model
 
     def transform(self, dataframe: pd.DataFrame) -> pd.DataFrame:
-        images = dataframe["images"]
-
-        results: t.List[pd.Series] = []
+        images = dataframe[["image"]]
+        images_series = dataframe["image"]
```
Comment (RobbeSneyders) [inline src/components/detect_text/src/main.py:165]: Can we work with just one of these?

**Review by RobbeSneyders — COMMENTED** [review]
Thanks @mrchtr!

File: src/components/add_clip_score/src/main.py (hunk 1 above)
Comment (mrchtr) [reply src/components/add_clip_score/src/main.py]: Wasn't working without the axis. We are operating row wise and not column wise. Following the documentation axis=1 defines the row wise operation.

File: src/pipeline.py [hunk 3]
```diff
@@ -3,108 +3,88 @@
...
-pipeline.add_op(load_from_hub_op)
-pipeline.add_op(download_images_op, dependencies=load_from_hub_op)
-pipeline.add_op(resize_images, dependencies=download_images_op)
-pipeline.add_op(detect_text_op, dependencies=resize_images)
-pipeline.add_op(mask_images_op, dependencies=detect_text_op)
-pipeline.add_op(embed_images_op, dependencies=mask_images_op)
-pipeline.add_op(add_clip_score_op, dependencies=embed_images_op)
-pipeline.add_op(filter_clip_score_op, dependencies=add_clip_score_op)
+filtered_clip_score_op = images_with_clip_score.apply(
+    "components/filter_clip_score", arguments={"threshold_score": 0.19}
```
Comment (PhilippeMoussalli) [inline src/pipeline.py:89]: nit: can you format the arguments on a new line?

File: src/components/mask_images/fondant_component.yaml [hunk 4]
```diff
@@ -1,21 +1,17 @@
 name: Mask images
 description: Component that masks images based on bounding boxes
-image: ghcr.io/ml6team/mask_images:0.6.2
+image: ghcr.io/ml6team/mask_images:0.8dev5
```
Comment (PhilippeMoussalli) [inline src/components/mask_images/fondant_component.yaml:3]: why are we using dev5?

File: src/components/filter_clip_score/fondant_component.yaml [hunk 5]
```diff
@@ -1,17 +1,17 @@
 name: Filter CLIP score
 description: Component that filters out bad captions (Empty captions, Captions with weird characters, Captions that are dates)
-image: ghcr.io/ml6team/filter_clip_score:0.6.2
+image: ghcr.io/ml6team/filter_clip_score:dev
```
Comment (PhilippeMoussalli) [inline src/components/filter_clip_score/fondant_component.yaml:3]: shouldn't this be 0.8?

File: src/components/detect_text/src/models/head/fast_head.py [hunk 6]
```diff
@@ -19,7 +19,7 @@
     from models.post_processing.ccl import ccl_cuda
 
     ccl_cuda_success = True
-    print("ccl_cuda successfully not installed!")
```
Comment (PhilippeMoussalli) [inline src/components/detect_text/src/models/head/fast_head.py:22]: LOL

**Pull request merged**